src/script/interpreter.cpp

```diff
@@ -1797,23 +1876,31 @@ bool SignatureHash(uint256 &sighashOut,
                                    sigHashType);
```

Comment:
The above needs:

```
if (!sigHashType.hasForkId() || !(flags & SCRIPT_ENABLE_SIGHASH_FORKID)) {
```

Reply:
We also need an error for:

```
This needs if (sigHashType.hasBIP341() && !sigHashType.hasForkId()) {
```

Currently.

Reply:
Too many tests currently rely on `0x20` being legacy

We can refactor that later.

For now, I'd just return false in `SigHashType::isDefined` in those cases.